core/src/main/scala/org/apache/spark/scheduler/ShuffleMapStage.scala

```diff
@@ -43,35 +43,53 @@ private[spark] class ShuffleMapStage(
     val shuffleDep: ShuffleDependency[_, _, _])
   extends Stage(id, rdd, numTasks, parents, firstJobId, callSite) {
 
+  private[this] var _mapStageJobs: List[ActiveJob] = Nil
```

Comment:
cc @mateiz can you explain when we can have more than one active jobs for a stage? Does this only happen if we do adaptive query execution and bypass the existing DAGScheduler to run shuffle map stages?